pkg/spoke/hubclientcert/controller.go

```diff
@@ -241,14 +244,15 @@ func (c *ClientCertForHubController) syncCSR(secret *corev1.Secret) (map[string]
 	// create a kubeconfig with references to the key/cert files in kubeconfigSecret if it dose not exists.
 	// So other components deployed in separated deployments are able to access this kubeconfig for hub as
 	// well by sharing the secret
+	var kubeconfigData = secret.Data[KubeconfigFile]
```

Comment:
this change is used to fix the current bug, for recovery invalid hub kubecofing test, the old code will miss the KubeconfigFile, because 
- at the first sync, the controller will create the hub kubeconfig secret with an invalid kubeconfig 
- at the second sync (recover the invalid hub kubecofing), because the secret already has the KubeconfigFile, the KubeconfigFile will miss